tdrs-backend/tdpservice/parsers/validators/base.py

```diff
@@ -0,0 +1,210 @@
+"""Base functions to be overloaded and composed from within the other validator classes."""
+
+from .util import _is_empty
+
+
+class ValidatorFunctions:
+    """Base higher-order validator functions that can be composed and customized."""
+
+    @staticmethod
+    def _handle_cast(val, cast):
+        return cast(val)
+
+    @staticmethod
+    def _handle_kwargs(val, **kwargs):
+        if 'cast' in kwargs and kwargs['cast'] is not None:
+            val = ValidatorFunctions._handle_cast(val, kwargs['cast'])
+
+        return val
+
+    @staticmethod
+    def _make_validator(func, **kwargs):
```

Comment:
suggestion:
If you change _make_validator to a decorator, then you can re-write for example the isEqual as follows:

```
@staticmethod
@validator 
def isEqual(option, **kwargs):
        """Return a function that tests if an input param is equal to option."""
        return lambda val: val == option

```

Reply:
made this change, think it cleans things up a lot! thank you!